### DISCUSSION TITLE
Bulk add interfaces to interface templates using API

### DISCUSSION THREAD

**Discussion by themmini** [netbox-community/netbox#9375] [Q&A] [answered]
Trying to get to grips with the API.
Is there a way to add many interfaces in 1 request using "PUT" to a device template?

Something like this - I've got it formatted as JSON from Postman which I will then eventually import to python

payload=
`{
    "device_type": 112,

    {
        "name": "NIC 1",
        "type": "25gbase-x-sfp28",
        "mgmt_only": false
    },
    {
        "name": "NIC 2",
        "type": "25gbase-x-sfp28",
        "mgmt_only": false
    }

}`

**Comment by AndMrzv** [comment, marked as answer]
I use POST requests for bulk creation (https://demo.netbox.dev/api/docs/). And specify everything that's needed (device_type in your case) inside every element.
